vpc/basic_firewall_rule/main.tf

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START compute_firewall_create]
+resource "google_compute_firewall" "rules" {
+  name        = "my-firewall-rule"
+  network     = "default"
+  description = "Creates firewall rule targeting tagged instances"
+
+  allow {
+    protocol = "tcp"
+    ports    = ["80", "8080", "1000-2000"]
```

Comment:
Can we match the behaviour of other code samples ([example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/5741faad622a6dbbb39e3c76526b17b32890be68/compute/client_library/snippets/firewall/create.py#L101))? We should open only ports 80 and 443.